### PR TITLE
Move Scripts to a sub-directory and enhance usage

### PR DESCRIPTION
**Pull request by hasezoey**
**Summary**

This PR does:
- move all scripts (like `website.js` and `static.js`) to a sub-directory (named `scripts`)
- update `website.js` and `static.js` to be used easier
  - update `website.js` to export its functions
  - update `static.js` to call `website.js`'s functions and also start the watch mode
  - update `package.json` to only call `static.js` for `docs:view`
- renames a variable from `jade` to `pug` to be not confusing
- removes some pug unused pug variables / options (like `marked` / `markedCode`)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,18 +77,18 @@
...
     "docs:checkout:gh-pages": "git checkout gh-pages",
     "docs:checkout:legacy": "git checkout 5.x",
-    "docs:generate": "node website.js",
+    "docs:generate": "node ./scripts/website.js",
     "docs:generate:search": "node docs/search.js",
     "docs:merge:stable": "git merge master",
     "docs:merge:legacy": "git merge 5.x",
     "docs:test": "npm run docs:generate && npm run docs:generate:search",
-    "docs:view": "node website.js && node static.js",
+    "docs:view": "node ./scripts/static.js",
```
Comment (Uzlopak) [inline package.json:85]: If you do this you probably have to modify a markdown file i think.

**Review by Uzlopak — CHANGES_REQUESTED** [review]
add `npm run docs:generate ` to this documentation

https://github.com/Automattic/mongoose/blob/master/CONTRIBUTING.md#documentation

File: package.json (hunk 1 above)
Comment (Uzlopak) [reply package.json:85]: see below

**Comment by hasezoey** [comment]
> add npm run docs:generate to this documentation

and what should it do? just call `website.js`?

**Comment by hasezoey** [comment]
> add npm run docs:generate to this documentation

what exactly should be added there and where? because currently the documentation (where `docs:view` is listed) says:

> If you'd like to preview your documentation changes, first commit your changes to your local master branch, then execute:

emphasis on `preview` (ie starting the webserver)

**Comment by Uzlopak** [comment]
Well if you dont generate than you dont see the result I think

**Comment by hasezoey** [comment]
> Well if you dont generate than you dont see the result I think

then please read what this PR does again:
> update website.js and static.js to be used easier 
> update static.js to call website.js's functions and also start the watch mode

**Comment by Uzlopak** [comment]
ah nice

**Comment by Uzlopak** [comment]
Yesterday when i tested it it seems that index.html was not send with the correct content encoding. I will test it tonight again

**Comment by hasezoey** [comment]
> Yesterday when i tested it it seems that index.html was not send with the correct content encoding. I will test it tonight again

weird, though i did not change anything regarding encoding that i know of, also which `index.html` are you meaning?

**Comment by hasezoey** [comment]
I just noticed that there is another script, but i am not sure about so i will do it in a later PR, named `docs/search.js` which is seemingly only meant to generate the search when deployed

File: docs/splitApiDocs.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 const api = require('./source/api');
 const fs = require('fs');
-const jade = require('pug');
```
Comment (vkarpov15) [inline docs/splitApiDocs.js:5]: :man_facepalming: thanks

**Review by vkarpov15 — APPROVED** [review]
Thanks :+1:

**Pull request merged**